tests/phpunit/tests/blocks/register.php

```diff
@@ -552,6 +552,132 @@ public function test_block_register_block_type_proxy_for_metadata() {
 		$this->assertSame( 'tests/notice', $result->name );
 	}
 
+	/**
+	 * Tests that an array value for 'editor_script' is correctly set and retrieved.
+	 *
+	 * As 'editor_script' is now a deprecated property, this should also set
+	 * the value for the 'editor_script_handles' property.
+	 *
+	 * @ticket 56707
+	 *
+	 * @covers ::register_block_type
+	 * @covers WP_Block_Type::__set
+	 * @covers WP_Block_Type::__get
+	 *
+	 * @dataProvider data_register_block_type_accepts_editor_script_array
+	 *
+	 * @param array $editor_script The editor script array to register.
+	 * @param array $expected      The expected registered editor script.
+	 */
+	public function test_register_block_type_accepts_editor_script_array( $editor_script, $expected ) {
+		$settings = array( 'editor_script' => $editor_script );
+		register_block_type( 'core/test-static', $settings );
+
+		$registry              = WP_Block_Type_Registry::get_instance();
+		$registered            = $registry->get_all_registered();
+		$actual_script         = $registered['core/test-static']->editor_script;
```

Comment:
There really needs to be a check to see if the property exists.

Reply:
Addressed in 45775e0.